### ISSUE TITLE
extensions.datacube.Dimension.dim_type return type?

### ISSUE THREAD

**Issue by soxofaan** [stac-utils/pystac#1368]
According to current docs `pystac.extensions.datacube.Dimension.dim_type` is a property that "returns" a `DimensionType` or `str`:

https://github.com/stac-utils/pystac/blob/630116c7c31639818b1342c9d2f4231a14dca4f0/pystac/extensions/datacube.py#L78

That's quite confusing to work with as a user. 
Say I want to check if a dimension is temporal, do I have to check if `dim_type` equals `DimensionType.TEMPORAL` or `"temporal"`?
As far as I understand, the `get_required` in `dim_type` will always return a string, so why is it annotated to possibly return a `DimensionType` as well?

**Comment by jsignell** [comment]
`DimensionType` is just a `StringEnum` which means that you can write comparisons with strings:

```python
>>> from pystac.extensions.datacube import DimensionType
>>> 
>>> DimensionType.SPATIAL == "spatial"
True
```

**Comment by soxofaan** [comment]
Indeed, I figured that out and that is handy, but is that recommended usage documented somewhere?
(Note that I created this ticket not only to figure out what to do, but also raise the issue that the documentation is currently not clear on this point)

First, I don't think the docs explicitly expose that `DimensionType` is an Enum subtype (I had to check the source to find that out) which is important to know how to properly work with it (e.g. the type relationship between the class and its attributes, and how to properly check equality).
Moreover (and I just learned about this by looking into this issue :smile: ) you are using this `StringEnum` subclass (backport of python 3.11 StrEnum I guess), which allows omitting `.value` to properly check equality. That is pretty confusing as every time I encounter enums in python, some kind of mantra kicks in to "never forget `.value`".

In short, I think, the current docs made it hard for me to figure out I just can do
```
dim.dim_type == DimensionType.TEMPORAL
```
to detect that a dimension is a temporal dimension

**Comment by jsignell** [comment]
Yeah that makes sense. If you feel like the docs could be clearer please feel free to open a PR.
